src/neo-vm/ExecutionEngine.cs

```diff
@@ -289,7 +295,42 @@ private bool ExecuteInstruction()
                     }
                 case OpCode.THROW:
                     {
-                        return false;
+                        if (!TryPop(out StackItem error)) return false;
+                        throw new CatcheableException(error);
```

Comment:
```suggestion
                        return HandleException(new CatcheableException(error));
```

If so, there is no need to throw the exception, right?

Reply:
I think it's better to throw the exception, as we may need to rethrow a exception. It may be clearer by using try-catch.

Reply:
But if you catch the exception, `PostExecuteInstruction()` won't be executed.

Reply:
If so, we need `FaultState`.

Reply:
Got,  we don't need `FaultState`, just by using `HandleException`.

Reply:
Done, the code looks more cleaner.  

Should I remove the `CatcheableException`, it seems we don't need it now.

Reply:
Yes, please. Let's remove it.